### PR TITLE
Add join argument allowing formating list of values separated by a

### PR DESCRIPTION
**Pull request by olivier80**
string. Each value is formated according the format specifier.

### CODE REVIEW

**Comment by vitaut** [comment]
Thanks for contributing. The PR looks good but could you follow the coding conventions described in https://github.com/fmtlib/fmt/blob/master/CONTRIBUTING.rst?

**Comment by olivier80** [comment]
Sorry. Hope it's ok now.

File: fmt/format.h [hunk 1]
```diff
@@ -3865,6 +3865,73 @@ void BasicFormatter<Char, AF>::format(BasicCStringRef<Char> format_str) {
   }
   write(writer_, start, s);
 }
+
+
+template <typename Char, typename It>
+struct ArgJoin
+{
```
Comment (vitaut) [inline fmt/format.h]: The brace belongs to the previous line. Otherwise looks good.

File: fmt/format.h [hunk 2]
```diff
@@ -3865,6 +3865,72 @@ void BasicFormatter<Char, AF>::format(BasicCStringRef<Char> format_str) {
   }
   write(writer_, start, s);
 }
+
+
```
Comment (vitaut) [inline fmt/format.h]: nit: extra empty line

File: fmt/format.h [hunk 3]
```diff
@@ -3865,6 +3865,72 @@ void BasicFormatter<Char, AF>::format(BasicCStringRef<Char> format_str) {
...
+}
+
+template <typename Range>
+auto join(const Range& range, const BasicCStringRef<wchar_t>& sep)
+    -> ArgJoin<wchar_t, decltype(std::begin(range))> {
+  return join(std::begin(range), std::end(range), sep);
+}
+#endif
+
+
```
Comment (vitaut) [inline fmt/format.h]: also here

File: fmt/format.h [hunk 4]
```diff
@@ -3865,6 +3865,72 @@ void BasicFormatter<Char, AF>::format(BasicCStringRef<Char> format_str) {
...
+      format_str = save;
+      f.format(format_str, internal::MakeArg<fmt::BasicFormatter<Char, ArgFormatter> >(*it++));
+    }
+  }
+  format_str = end + 1;
+}
+
+
+
+
```
Comment (vitaut) [inline fmt/format.h]: and a bunch here

**Comment by vitaut** [comment]
Merged, thanks!

**Comment by olivier80** [comment]
Thank you !

**Pull request merged**